src/utils/utils.js

```diff
@@ -0,0 +1,60 @@
+export const generateNumeratorCode = (numerators) => {
+    const lastCode = numerators[numerators.length - 1].code;
+    const lastNumber = parseInt(lastCode.slice(1));
+
+    const newCodeNumber = lastNumber + 1;
+    const newCode = "C" + newCodeNumber;
+
+    return newCode;
+}
+
+// TODO: setup the global context to get the stored configurations
+export const getConfigObjectsForAnalytics = (configurations, groupCode) => {
+    
+     // Find the group with the given code
+     const group = configurations.groups.find((g) => g.code === groupCode);
+     if (!group) {
+        // Return an empty array if the group is not found
+        return [];
+    }
+
+    // Get members of the group
+    const members = group.members;
+
+    // Initialize a Set to store unique dataset IDs
+    const uniqueDatasetIDs = new Set();
+  
+    // Filter numerators with dataID not null and in the group's members
+    const numeratorsInGroup = configurations.numerators.filter((numerator) => {
+        return members.includes(numerator.code) && numerator.dataID !== null;
```

Comment:
I think there needs to be a condition to check if the dataSetID on the numerator is not null as well? (I think we cannot pull relevant data unless the data element has both a data set and indicator). This would also mean that you don't have to check for null values when mapping numeratorsInGroup